File: elixir/fizzbuzz.exs
fizzbuzz = fn (_, 0, 0) -> "fizzbuzz"
              (_, 0, _) -> "fizz"
              (_, _, 0) -> "buzz"
              (n, _, _) -> n
end

fizzbuzz = fn(n) -> fizzbuzz.(n, rem(n, 3), rem(n, 5)) end

Enum.map 1..100, fn(i) -> IO.puts(fizzbuzz.(i)) end

File: go/fizzbuzz.go
package main

import "fmt"

func main() {
  for i := 1; i <= 100; i++ {
    fizz := ""    
    buzz := ""

    if i % 3 == 0 {
      fizz = "fizz"
    }
   
    if i % 5 == 0 {
      buzz = "buzz"
    }

    if fizz != "" || buzz != "" {
      fmt.Println(fizz + buzz)
    } else {
      fmt.Println(i)
    }
  }  
}
